### ISSUE TITLE
08-k-nearest-neighbours-winequality-red.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/josefina-aispuro-merelles-machine-learning#4]
Hola @josefina-aispuro-merelles , por aquí las issues!

## LO QUE HAS HECHO MUY BIEN

- **Imports completos**: Todo lo necesario incluyendo `StandardScaler`, métricas y pickle.

- **Análisis descriptivo completo**: shape, info(), describe() - Muy bien documentado.

- **Limpieza de duplicados**: Excelente que verificaste y eliminaste los 240 duplicados.

- **Visualización**: Histogramas y heatmap de correlaciones. Muy buen análisis exploratorio.

- **Observaciones del EDA**: Identificaste las correlaciones más fuertes y la relación de alcohol con calidad (0.48).

- **Creación de la variable label**: Correcto el agrupamiento en 3 clases (0, 1, 2).

- **Escalado con StandardScaler**: ¡Perfecto! Esto es **fundamental** para KNN ya que usa distancias.

- **Optimización con GridSearchCV**: Probaste `n_neighbors`, `weights` y `metric`. Muy completo.

- **Guardaste el modelo optimizado**

- **Mejora documentada**: De 81.98% a 86.02% - buena observación.

---

### **1. Error de copy-paste en observaciones**
```markdown
The decision tree model achieved 81.98% accuracy on the test set.
```

No estás usando Decision Tree, estás usando **KNN**. **Este error también apareció en tu ejercicio de Naive Bayes!!!!!**

---

### **2. NO completaste el requisito de graficar accuracy vs k**
El enunciado pedía específicamente:
> *"Crea un bucle para probar diferentes valores de k (de 1 a 20). Guarda los resultados en una lista. Grafica accuracy vs k para encontrar el mejor valor."*

Solo usaste GridSearchCV, pero faltó el bucle manual con la gráfica. Ejemplo:

```python
k_values = range(1, 21)
accuracies = []

for k in k_values:
    model = KNeighborsClassifier(n_neighbors=k)
    model.fit(X_train_scaled, y_train)
    y_pred = model.predict(X_test_scaled)
    accuracies.append(accuracy_score(y_test, y_pred))

plt.plot(k_values, accuracies, marker='o')
plt.xlabel('k')
plt.ylabel('Accuracy')
plt.title('Accuracy vs k')
plt.show()
```

---

### **3. NO usaste confusion_matrix ni classification_report**
El enunciado pedía evaluar con:
- `accuracy_score` 👍 
- `confusion_matrix` 👎 
- `classification_report` 👎 

Deberías haber añadido:
```python
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
```

---

### **4. Doble fit en el scaler (redundante)**
```python
scaler.fit(X_train)
X_train_scaled = scaler.fit_transform(X_train)  # ← fit_transform ya incluye fit
```

Puedes usar solo `fit_transform` en train y `transform` en test:
```python
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)
```

---

**Buen trabajo, Josefina.** El notebook está bien estructurado y el análisis exploratorio es muy completo. Sin embargo, te han faltado algunos requisitos del enunciado: la gráfica de accuracy vs k, las métricas adicionales.. ¡Buen trabajo en general! Cierra la issue cuando esté!
